catalog/internal/utils.go

```diff
@@ -344,7 +344,11 @@ func CreateViewMetadata(
 	if err != nil {
 		return "", fmt.Errorf("failed to create view metadata file: %w", err)
 	}
-	defer out.Close()
+	defer func() {
+		if cerr := out.Close(); cerr != nil {
+			err = fmt.Errorf("error closing FileWriter: %w", cerr)
```

Comment:
if `err` is already `!= nil`, should we return both errors?

Reply:
Yeah, makes sense, changing to `errors.Join`